agent/exec/container/controller.go

```diff
@@ -395,7 +395,65 @@ func (r *controller) Remove(ctx context.Context) error {
 	return nil
 }
 
+// waitReady waits for a container to be "ready".
+// Ready means it's past the started state.
+func (r *controller) waitReady(pctx context.Context) error {
+	if err := r.checkClosed(); err != nil {
+		return err
+	}
+
+	ctx, cancel := context.WithCancel(pctx)
+	defer cancel()
```

Comment:
What's the cancel useful for? Do any of the function calls below spawn goroutines that outlive this function?